cmake-tool/helpers/rootserver.cmake

```diff
@@ -101,37 +97,30 @@ function(DeclareRootserver rootservername)
             else()
                 set(march rv64imafdc)
             endif()
-            if(UseRiscVBBL)
-                # Package up our final elf image into the Berkeley boot loader.
-                # The host string is extracted from the cross compiler setting
-                # minus the trailing '-'
+            if(UseRiscVOpenSBI)
+                # Package up our final elf image into OpenSBI.
                 if("${CROSS_COMPILER_PREFIX}" STREQUAL "")
                     message(FATAL_ERROR "CROSS_COMPILER_PREFIX not set.")
                 endif()
 
-                string(
-                    REGEX
-                    REPLACE
-                        "^(.*)-$"
-                        "\\1"
-                        host
-                        "${CROSS_COMPILER_PREFIX}"
-                )
-                get_filename_component(host ${host} NAME)
+                if("${KernelOpenSBIPlatform}" STREQUAL "")
+                    message(FATAL_ERROR "KernelOpenSBIPlatform not set.")
+                endif()
+
                 file(GLOB_RECURSE deps)
+                set(
+                    OPENSBI_FW_PAYLOAD_ELF
+                    "${OPENSBI_PATH}/build/platform/${KernelOpenSBIPlatform}/firmware/fw_payload.elf"
+                )
                 add_custom_command(
-                    OUTPUT "${CMAKE_BINARY_DIR}/bbl/bbl"
-                    COMMAND mkdir -p ${CMAKE_BINARY_DIR}/bbl
+                    OUTPUT "${OPENSBI_FW_PAYLOAD_ELF}"
                     COMMAND
-                        cd ${CMAKE_BINARY_DIR}/bbl && ${BBL_PATH}/configure
-                        --quiet
-                        --host=${host}
-                        --with-arch=${march}
-                        --with-payload=${elf_target_file}
-                            && make -s clean && make -s > /dev/null
-                    DEPENDS ${elf_target_file} elfloader ${USES_TERMINAL_DEBUG}
+                        cd "${OPENSBI_PATH}" && ${CMAKE_OBJCOPY} -O binary "${elf_target_file}"
+                        "payload" && make -s clean && CROSS_COMPILE=${CROSS_COMPILER_PREFIX} make
+                        PLATFORM="${KernelOpenSBIPlatform}" FW_PAYLOAD_PATH=payload
```

Comment:
It's not necessary to build in the source directory.
The OpenSBI Makefile defines a variable "O" to set the build directory and the payload binary can be created in the build directory.

Reply:
great, that will be very useful. Can you propose a Patch for this?

Reply:
https://github.com/seL4-drone-project/seL4_tools/commit/6f8814ee0950e2ebf989de16e2b801ce71eec4e9

Reply:
@alistair23: I'd leave it to you to have a look at the changed from @alexf91, check if the work for you also then merge them into this PR. Otherwise, we'll see that we get your PR finally merged first and then have @alexf91 make a PR on top of this.

Reply:
I have changed this to use the "O" variable.

This should be ready to merge now.